src/audio/play.rs

```diff
@@ -0,0 +1,26 @@
+//! Provides functions used to play audio.
+
+use rodio::*;
+
+use std::io::Cursor;
+
+use super::source::Source;
+use super::output::Output;
+use super::DecoderError;
+
+/// Play a sound once.
+///
+/// This will return an Error if the loaded audio file in source could not be decoded.
+pub fn once(source: &Source, endpoint: &Output) -> Result<(), DecoderError> {
```

Comment:
Calling this function from ECS is a bit ugly, since you got two unwraps above and the default ouptut has to be retrieved every time. Not sure, but I don't think a game should really crash / exit even if such a thing failed. Especially during development, it's enough to see an error in the console IMO.